### PR TITLE
Simplified on_article_delete a bit. Added tests.

### PR DESCRIPTION
**Pull request by atombrella**

### CODE REVIEW

File: tests/core/test_signals.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        )
+
+    @override_settings(LOST_AND_FOUND_SLUG='lost')
+    def test_delete_root_with_settings(self):
+        with mock.patch('wiki.models.urlpath.on_article_delete', autospec=True) as signal:
+            pre_delete.connect(signal, sender=Article)
+            self.fake_root_article.delete()
+        self.assertEqual(signal.call_count, 1)
+        # self.assertEqual(self.a1.urlpath_set.all().count(), 1)
+        print(self.a1.urlpath_set.all(), self.a2.urlpath_set.all())
```
Comment (benjaoming) [inline tests/core/test_signals.py]: It looks like a debug statement and a commented out assertion?

File: tests/core/test_signals.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    # @override_settings(LOST_AND_FOUND_SLUG=None)
+    # def test_delete_root(self):
+    #     self.sub_urlpath.delete()
+    #     self.assertEqual(self.a1)
```
Comment (benjaoming) [inline tests/core/test_signals.py]: Is this intentionally commented out?

File: tests/core/test_signals.py (hunk 1 above)
Comment (atombrella) [reply tests/core/test_signals.py]: Yes. The WIP label should hint at something. I'm trying to figure out how to get it running. Signals testing seems a bit tricky.

Comment (benjaoming) [reply tests/core/test_signals.py]: Sorry, I didn't know that label, so kind of over-looked it :)

File: tests/core/test_signals.py (hunk 2 above)
Comment (benjaoming) [reply tests/core/test_signals.py]: @atombrella do you recall why this was commented out?

File: tests/core/test_signals.py (hunk 1 above)
Comment (atombrella) [reply tests/core/test_signals.py]: You can create labels. They are not predefined by Github.

File: tests/core/test_signals.py (hunk 2 above)
Comment (atombrella) [reply tests/core/test_signals.py]: I had trouble to get it to pass. I'm not sure if it's worth keeping this PR open until this is sorted out. We had a chat about the things that this PR touches in person a couple of months ago.

Comment (benjaoming) [reply tests/core/test_signals.py]: Ah okay, yeah, I kind of recall that, thanks :)

**Pull request closed without merging**